### PR TITLE
TS fixes

### PR DESCRIPTION
**Pull request by thompsongl**
This was easier than describing the changes 😄

### CODE REVIEW

File: src/components/steps/index.ts [hunk 1]
```diff
@@ -5,3 +5,5 @@ export { EuiSteps } from './steps';
 export { EuiSubSteps } from './sub_steps';
 
 export { EuiStepsHorizontal } from './steps_horizontal';
+
+export { EuiStepStatus } from './step_number';
```
Comment (thompsongl) [inline src/components/steps/index.ts:9]: Kibana imports and uses this type directly

File: src/components/steps/step.tsx [hunk 2]
```diff
@@ -44,16 +44,13 @@ export const EuiStep: FunctionComponent<StandaloneEuiStepProps> = ({
   return (
     <div className={classes} {...rest}>
       <div className="euiStep__titleWrapper">
-        {/* // EuiI18n has trouble with the string setting
-        // @ts-ignore */}
         <EuiI18n
           token="euiStep.ariaLabel"
-          default={({ status }) => {
+          default={({ status }: { status?: EuiStepStatus }) => {
```
Comment (thompsongl) [inline src/components/steps/step.tsx:49]: If I'm honest, I have no idea why this resolves the errors

File: src/components/steps/step_horizontal.test.tsx [hunk 3]
```diff
@@ -63,27 +61,27 @@ describe('EuiStepHorizontal', () => {
 
     describe('onClick', () => {
       test('is called when clicked', () => {
-        const onClickHandler = sinon.stub();
+        const onClickHandler = jest.fn();
```
Comment (thompsongl) [inline src/components/steps/step_horizontal.test.tsx:64]: Just bypassing the need to import and `@ts-ignore` `sinon`

**Review by cchaos — APPROVED** [review]
Thanks!!

**Pull request merged**